spec/datadog/appsec/extensions_spec.rb

```diff
@@ -60,9 +60,7 @@
 
       describe '#enabled=' do
         subject(:enabled_) { settings.enabled = false }
-        it do
-          expect { enabled_ }.to change { settings.enabled }.from(true).to(false)
-        end
+        it { expect { enabled_ }.to change { settings.enabled }.from(true).to(false) }
       end
 
```

Comment:
⬆️  I can't comment on the exact lines above, but this whole spec seems weird.

This is not a spec for `Datadog::Core::Configuration::Settings` as the describe claims, this is a spec for `Settings.new.appsec`.

And... it quite repeats the new `spec/datadog/appsec/configuration/settings_spec.rb`. My suggestion would be to avoid doing so -- by having this spec check that `Settings.new.appsec` correctly delegates to a `Datadog::AppSec::Configuration::Settings` but not actually testing it through here.

That, or perhaps removing the `Datadog::AppSec::Configuration::Settings` and testing it completely through here. Having a spec that is almost copy-pasted from the other seems weird...?

Reply:
> this whole spec seems weird

Yes it is. @marcotc added the extension spec to ensure that propagation of these values happens correctly but the `subject` being `Settings.new.appsec` defeats the purpose. I chose to implement a spec for `Datadog::AppSec::Configuration::Settings` because it felt more logical/unit to have these examples on the instance itself instead of going through and depending on a bigger chunk of code.

Interestingly enough the bug I mentioned in another comment is that `reset!` doesn't quite work because there's some memoization happening somewhere in or around that bigger chunk of code that escapes reset attempts and adding more test cases makes them fail randomly depending on ordering for reasons that only happen in specs.

> My suggestion would be to avoid doing so -- by having this spec check that Settings.new.appsec correctly delegates to a Datadog::AppSec::Configuration::Settings but not actually testing it through here.

Agreed.